src/rust/engine/src/nodes.rs

```diff
@@ -480,14 +480,13 @@ impl Snapshot {
   }
 
   pub fn lift_path_globs(item: &Value) -> Result<PathGlobs, String> {
-    let include = externs::project_multi_strs(item, "include");
-    let exclude = externs::project_multi_strs(item, "exclude");
+    let globs = externs::project_multi_strs(item, "globs");
 
-    let origin = externs::project_str(item, "description_of_origin");
-    let description_of_origin = if origin.is_empty() {
+    let description_of_origin_field = externs::project_str(item, "description_of_origin");
+    let description_of_origin = if description_of_origin_field.is_empty() {
       None
     } else {
-      Some(origin)
+      Some(description_of_origin_field)
```

Comment:
Note: we could eventually consider putting this "`None` if empty string" logic some into some string extension trait that we could import and then call like:
```python
let description_of_origin = externs::project_str(item, "description_of_origin").into_empty_option();
```

Reply:
Not super justifiable to make right now, though. Just good to keep track if we start doing this elsewhere.

Reply:
Yes! Ideally the Python dataclass could have `Optional[str]`. I encoded it as always being `str` and am overloading the empty string `""` to symbolize `None` for easier interop with Rust. That's not great.